homeassistant/components/zeroconf/discovery.py

```diff
@@ -336,6 +338,15 @@ def _async_process_service_update(
             return
         _LOGGER.debug("Discovered new device %s %s", name, info)
         props: dict[str, str | None] = info.properties
+
+        # Instance ID conflict detection for Home Assistant core
+        if service_type == ZEROCONF_TYPE:
+            discovered_instance_id = props.get("uuid")
+            if discovered_instance_id:
+                self.hass.async_create_task(
+                    self._async_check_instance_id_conflict(discovered_instance_id, info)
+                )
+
```

Comment:
I think there might be a risk of a thundering heard of tasks. Can we cache local_instance_id and local_address and make _async_check_instance_id_conflict synchronous?

Reply:
Actually we should have our zeroconf ServiceInfo generated at startup time. We could cache that and compare it against what was discovered